### ISSUE TITLE
[Defactor] Support a `Mask.where` which acts like `jax.numpy.where`, but for `Mask` values

### ISSUE THREAD

**Issue by femtomc** [genjax-community/genjax#1271]

**Comment by sritchie** [comment]
@mccoybecker, are you thinking of something like

```
    def where(self, condition: ScalarBool) -> "Mask[R]":
        return Mask(self.flag.and_(Flag(condition)), self.value)
```

inside `Mask`?

**Comment by sritchie** [comment]
@femtomc can you say more about what you want here?

**Comment by femtomc** [comment]
No (in response to reply above last reply), but perhaps `Mask.get` is a better name -- like `jnp.where(flag, arr1, arr2)` chooses between the two arrays -- here, if I have a mask in hand:
```python
mask.get(fallback_arr)
```

The point being that either we unwrap the mask (if valid), or we return the default fallback value.

**Comment by sritchie** [comment]
Got it, let's call it `get_or_else`.
